rskj-core/src/main/java/co/rsk/cli/migration/UnitrieMigrationTool.java

```diff
@@ -0,0 +1,395 @@
+ */
+public class UnitrieMigrationTool {
+
+    private final KeyValueDataSource orchidContractDetailsDataStore;
+    private final KeyValueDataSource orchidContractsStorage;
+    private final TrieStore orchidContractsTrieStore;
+    private final TrieStore orchidAccountsTrieStore;
+    private final String databaseDir;
+    private final Map<RskAddress, TrieStore> contractStoreCache = new HashMap<>();
+    private final Map<ByteArrayWrapper, RskAddress> addressHashes;
+    private final TrieConverter trieConverter;
+    private final Map<Keccak256, DataWord> keccak256Cache;
+    private final Block blockToMigrate;
+    private final Repository unitrieRepository;
+    private final StateRootHandler stateRootHandler;
+
+    public UnitrieMigrationTool(Block blockToMigrate,
+                                String databaseDir,
+                                Repository unitrieRepository,
+                                StateRootHandler stateRootHandler,
+                                TrieConverter trieConverter) {
+        this.databaseDir = databaseDir;
+        this.blockToMigrate = blockToMigrate;
+        this.unitrieRepository = unitrieRepository;
+        this.stateRootHandler = stateRootHandler;
+        this.trieConverter = trieConverter;
+
+        this.orchidContractDetailsDataStore = RskContext.makeDataSource("details", databaseDir);
+        this.orchidContractsStorage = RskContext.makeDataSource("contracts-storage", databaseDir);
+        this.orchidContractsTrieStore = new CachedTrieStore(new TrieStoreImpl(orchidContractsStorage));
+        this.orchidAccountsTrieStore = new CachedTrieStore(new TrieStoreImpl(RskContext.makeDataSource("state", databaseDir)));
+        this.keccak256Cache = new HashMap<>();
+        this.addressHashes = orchidContractDetailsDataStore.keys().stream()
+                .filter(accountAddress -> accountAddress.length == 20)
+                .collect(
+                        Collectors.toMap(accountAddress -> ByteUtil.wrap(Keccak256Helper.keccak256(accountAddress)),
+                                RskAddress::new
+                        )
+                );
+        this.addressHashes.put(ByteUtil.wrap(Keccak256Helper.keccak256(PrecompiledContracts.REMASC_ADDR.getBytes())), PrecompiledContracts.REMASC_ADDR);
+        this.addressHashes.put(ByteUtil.wrap(Keccak256Helper.keccak256(RemascTransaction.REMASC_ADDRESS.getBytes())), RemascTransaction.REMASC_ADDRESS);
+    }
+
+    public void migrate() {
+        Trie migratedTrie = migrateState(blockToMigrate);
+        unitrieRepository.flush();
+
+        stateRootHandler.register(
+                blockToMigrate.getHeader(),
+                migratedTrie
+        );
+    }
+
+    private Trie migrateState(Block blockToMigrate) {
+        byte[] orchidStateRoot = blockToMigrate.getStateRoot();
+        System.out.printf("====== %07d (%s) =======\n", blockToMigrate.getNumber(), Hex.toHexString(orchidStateRoot));
```

Comment:
I think these should all be `logger.info`s. Our nodes are usually deployed through `PPA`, and you never see the service's stdout.

Reply:
You're correct. I just fixed it and made all `System.out` to logger